test/widgets/message_list_test.dart

```diff
@@ -484,6 +487,22 @@ void main() {
 
       debugNetworkImageHttpClientProvider = null;
     });
+
+    testWidgets('Bot user is distinguished by showing an icon', (tester) async {
+      prepareBoringImageHttpClient();
+
+      final users = List.generate(3, (index) => eg.user(isBot: index == 0));
+
+      await setupMessageListPage(
+        tester,
+        messages: users.map((user) => eg.streamMessage(sender: user)).toList(),
+        users: users,
+      );
+
+      tester.widget(find.byIcon(ZulipIcons.bot));
```

Comment:
Can you check specifically that the icon appears next to the sender that is a bot, and also that an icon _doesn't_ appear on the senders that are not bots?

Reply:
Sure thing!